### PR TITLE
Adding revocations and precursors properties to oncotree nodes.

### PR DESCRIPTION
**Pull request by mandawilson**

### CODE REVIEW

File: core/src/main/java/org/mskcc/oncotree/topbraid/OncoTreeVersionRepository.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
     private String query = "PREFIX oncotree-version:<http://data.mskcc.org/ontologies/oncotree_version/> " +
         "SELECT ?api_identifier ?graph_uri ?description ?release_date ?visible " +
         "WHERE { " +
-        "   GRAPH <urn:x-evn-master:oncotree_version> { " +
+        "   GRAPH <urn:x-evn-master:oncotree_version_testing> { " +
```
Comment (mandawilson) [inline core/src/main/java/org/mskcc/oncotree/topbraid/OncoTreeVersionRepository.java]: TODO this must be changed back.

File: core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java [hunk 2]
```diff
@@ -306,6 +344,7 @@ private TumorType initTumorType(OncoTreeNode oncoTreeNode, Version version) thro
         tumorType.setName(oncoTreeNode.getName());
         tumorType.setColor(oncoTreeNode.getColor());
         tumorType.setParent(oncoTreeNode.getParentCode());
+        // do not copy history, rescinds, or precursors
```
Comment (sheridancbio) [inline core/src/main/java/org/mskcc/oncotree/utils/TumorTypesUtil.java]: we should add a reason with this comment

File: core/src/test/java/org/mskcc/oncotree/utils/CacheUtilTest.java [hunk 3]
```diff
@@ -61,6 +60,12 @@
     
     @Resource(name="oncoTreeAdditionalVersionRepositoryMockResponse")
     private List<Version> oncoTreeAdditionalVersionRepositoryMockResponse;
+
+    @Resource(name="legacyVersion")
+    private Version legacyVersion;
+
+    @Resource(name="latestVersion")
```
Comment (sheridancbio) [inline core/src/test/java/org/mskcc/oncotree/utils/CacheUtilTest.java]: we will probably need additional versions (at least one more) to test the propagation/accumulation of history over time.

File: core/src/test/java/org/mskcc/oncotree/utils/CacheUtilTest.java [hunk 4]
```diff
@@ -128,4 +133,105 @@ public void testRecacheWithBrokenOncoTreeVersionRepository() {
         config.resetNotWorkingVersionRepository(oncoTreeVersionRepository);
         cacheUtil.resetCache();
     }   
+
+    private void assertNoHistoryInChildren(Map<String, TumorType> tumorTypes) {
```
Comment (sheridancbio) [inline core/src/test/java/org/mskcc/oncotree/utils/CacheUtilTest.java]: these three functions are very similar. we could make a more general assert which takes the field to be tested as a parameter. (same comment for the assertXInChildren functions)

File: core/src/test/java/org/mskcc/oncotree/utils/CacheUtilTest.java [hunk 5]
```diff
@@ -128,4 +133,105 @@ public void testRecacheWithBrokenOncoTreeVersionRepository() {
...
+
+    @Test
+    public void testGetTumorTypesByVersion() {
+        Map<String, TumorType> returnedTumorTypes = cacheUtil.getTumorTypesByVersion(legacyVersion);
+        // there is only one node in returnedTumorTypes, and it is the TISSUE node, everything else is a child of that node
+        String rootTumorTypeCode = returnedTumorTypes.keySet().iterator().next();
+        TumorType rootTumorType = returnedTumorTypes.get(rootTumorTypeCode);
+        assertEquals("TISSUE", rootTumorTypeCode);
+        assertEquals("TISSUE", rootTumorType.getCode());
+        assertEquals(32, rootTumorType.getChildren().size());
```
Comment (sheridancbio) [inline core/src/test/java/org/mskcc/oncotree/utils/CacheUtilTest.java]: 32 could be a parameter defined in the config class (or wherever the mock data is) .. or be computed there.
alternatively, we could have a tree traversal to see if the tree returned matches the tree defined in our mock

File: core/src/test/java/org/mskcc/oncotree/utils/OncotreeTestConfig.java [hunk 6]
```diff
@@ -202,20 +214,51 @@ private Version setupMockVersion() {
...
             nextNode.setMainType(rawTestValueSource[pos + 2]);
             nextNode.setColor(rawTestValueSource[pos + 3]);
             nextNode.setParentCode(rawTestValueSource[pos + 4]);
+            nextNode.setURI(rawTestValueSource[pos + 5]);
             tmpOncoTreeRepositoryMockResponse.add(nextNode);
         }
         return tmpOncoTreeRepositoryMockResponse;
     }
 
+    private List<OncoTreeNode> setupOncotreeRepositoryMockResponseHistoryChange() {
```
Comment (sheridancbio) [inline core/src/test/java/org/mskcc/oncotree/utils/OncotreeTestConfig.java]: we may want to create a general function for applying sets of modifications at particular timepoints. The destination version could be an argument to this function as well as the prior set of oncotree nodes, and only the changes intended to happen at that point in time would be applied to construct the new version.

**Comment by angelicaochoa** [comment]
@mandawilson tested with importer and import types of cancer. No issues

File: core/src/main/java/org/mskcc/oncotree/topbraid/OncoTreeVersionRepository.java (hunk 1 above)
Comment (mandawilson) [reply core/src/main/java/org/mskcc/oncotree/topbraid/OncoTreeVersionRepository.java]: Done.

**Pull request merged**